torch/csrc/jit/tensorexpr/kernel.cpp

```diff
@@ -954,8 +955,85 @@ void TensorExprKernel::LowerToBackend(BackendType backend_type) {
   codegen_ = CreateCodeGen(codegen_name, stmt, params);
 }
 
+template <typename T>
+static bool isValidPrimProperty(const c10::optional<T>& a, T b) {
+  return !a.has_value() || *a == b;
+}
+
+static bool isValidVaryingShape(
+    const c10::VaryingShape& vs,
+    at::IntArrayRef sz) {
+  if (!vs.size().has_value()) {
+    // TODO: does it make sense to have kernels with completely unspecified
+    // shapes/strides
+    return true;
+  }
+
+  if (*vs.size() != sz.size()) {
+    return false;
+  }
+
+  for (size_t i = 0; i < sz.size(); i++) {
+    if (!isValidPrimProperty(vs[i], sz[i])) {
+      return false;
+    }
+  }
+  return true;
+}
+
 void TensorExprKernel::PickAndCheckBackendType(
     const at::ArrayRef<IValue>& inputs) {
+  TORCH_INTERNAL_ASSERT(
```

Comment:
Could you possibly pull this out into a helper function so it doesn't impede the readability of the "main" path?

Reply:
Will do!